webapp/components/user_settings/user_settings_security/user_settings_security.jsx

```diff
@@ -929,6 +966,368 @@ export default class SecurityTab extends React.Component {
+                        id='user.settings.tokens.confirmCreateTitle'
+                        defaultMessage='Create System Admin User Access Token'
+                    />
+                ),
+                confirmMessage: (
+                    <div className='alert alert-danger'>
+                        <FormattedHTMLMessage
+                            id='user.settings.tokens.confirmCreateMessage'
+                            defaultMessage='You are generating a user access token with System Admin permissions. Are you sure want to create this token?'
+                        />
+                    </div>
+                ),
+                confirmButton: (
+                    <FormattedMessage
+                        id='user.settings.tokens.confirmCreateButton'
+                        defaultMessage='Yes, Create'
+                    />
+                ),
+                confirmComplete: () => {
+                    this.handleCreateToken();
+                    trackEvent('settings', 'system_admin_create_user_access_token');
+                }
+            });
+
+            return;
+        }
+
+        this.handleCreateToken();
+    }
+
+    saveTokenKeyPress = (e) => {
+        if (e.which === Constants.KeyCodes.ENTER) {
+            this.confirmCreateToken();
+        }
+    }
+
+    confirmRevokeToken = (tokenId) => {
+        const token = this.props.userAccessTokens[tokenId];
+
+        this.setState({
+            showConfirmModal: true,
+            confirmTitle: (
+                <FormattedMessage
+                    id='user.settings.tokens.confirmDeleteTitle'
+                    defaultMessage='Delete {name} Token?'
+                    values={{
+                        name: token.description
+                    }}
+                />
+            ),
+            confirmMessage: (
+                <div className='alert alert-danger'>
+                    <FormattedHTMLMessage
+                        id='user.settings.tokens.confirmDeleteMessage'
+                        defaultMessage='Any integrations using this token will no longer be able to access the Mattermost API. You cannot undo this action. Are you sure want to delete this token?'
+                    />
+                </div>
+            ),
+            confirmButton: (
+                <FormattedMessage
+                    id='user.settings.tokens.confirmDeleteButton'
+                    defaultMessage='Yes, Delete'
+                />
+            ),
+            confirmComplete: () => {
+                this.revokeToken(tokenId);
+                trackEvent('settings', 'revoke_user_access_token');
+            }
+        });
+    }
+
+    revokeToken = async (tokenId) => {
+        const {error} = await this.props.actions.revokeUserAccessToken(tokenId);
+        if (error) {
+            this.setState({serverError: error.message});
+        }
+        this.handleCancelConfirm();
+    }
+
+    createTokensSection = () => {
```

Comment:
I'm ok with this but maybe is best if we split this into a different component? perhaps in a later PR?